src/kibana/components/vislib/lib/x_axis.js

```diff
@@ -91,14 +91,23 @@ define(function (require) {
      */
     XAxis.prototype._getXExtents = function (data, extent) {
       var ordered = this.ordered;
-
-      return d3[extent](data, function (d) {
-        if (extent === 'max') {
-          return d + ordered.interval;
+      var opts = [ordered[extent]];
+
+      var point = d3[extent](data);
+      if (extent === 'max') {
+        if (ordered.date) {
+          point = moment(point).add(ordered.interval).valueOf();
+        } else {
+          point += ordered.interval;
         }
+      }
+      opts.push(point);
 
-        return d;
-      });
+      return d3[extent](opts.reduce(function (opts, v) {
+        if (!_.isNumber(v)) v = +v;
+        if (!isNaN(v)) opts.push(v);
+        return opts;
+      }, []));
```

Comment:
This integrates `ordered.min` and `ordered.max` into the calculation of the X extents. It also uses `moment` to combine dates/intervals.

By using moment, adding ms + "yearly interval" works properly.